spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/util/DeploymentPropertiesUtils.java

```diff
@@ -34,14 +39,16 @@
  */
 public final class DeploymentPropertiesUtils {
 
+	private static final Logger logger = LoggerFactory.getLogger(DeploymentPropertiesUtils.class);
+
 	private DeploymentPropertiesUtils() {
 		// prevent instantiation
 	}
 
 	/**
 	 * Pattern used for parsing a String of comma-delimited key=value pairs.
 	 */
-	private static final Pattern DEPLOYMENT_PROPERTIES_PATTERN = Pattern.compile(",\\s*app\\.[^\\.]+\\.[^=]+=");
+	private static final Pattern DEPLOYMENT_PROPERTIES_PATTERN = Pattern.compile(",\\s*(app|deployer)\\.[^\\.]+\\.[^=]+=");
```

Comment:
I agree the parsing code has always been brittle. IMHO, we should first split on `,` (with support for some kind of escape mechanism, _e.g._ `\,`) and **then** look for `app.x` and `deployer.x`.

This can (and should) be the subject of another issue/PR though